stm32-modules/heater-shaker/firmware/host_comms_task/freertos_comms_task.cpp

```diff
@@ -308,3 +333,57 @@ static auto CDC_Receive(uint8_t *Buf, uint32_t *Len) -> int8_t {
     USBD_CDC_ReceivePacket(&_local_task.usb_handle);
     return USBD_OK;
 }
+
+void HAL_UARTEx_RxEventCallback(UART_HandleTypeDef *UartHandle, uint16_t Size) {
+    using namespace host_comms_control_task;
+    uint8_t *Buf =
+        reinterpret_cast<uint8_t *>(_local_task.committed_uart_rx_buf_ptr);
+    ssize_t Len = (ssize_t)(Size);
+    ssize_t remaining_buffer_count = (ssize_t)(UART_BUFFER_MAX_SIZE)-Len;
```

Comment:
If this represents the remaining contents of the buffer, shouldn't it be `_local_task.committed() + UART_BUFFER_MAX_SIZE - (_local_task.committed_uart_rx_buf_ptr + Size)`? As it is, it only provides the total size of the UART buffer less the number of bytes _in this Rx callback_